CMakeLists.txt

```diff
@@ -208,12 +218,17 @@ add_custom_command(
 #----------------------------------------------------------------------
 # Add JSON parser
 #----------------------------------------------------------------------
-add_subdirectory(third_party/json ${CMAKE_CURRENT_BINARY_DIR}/json EXCLUDE_FROM_ALL)
+if(CMAKE_CROSSCOMPILING AND NOT _GRPC_DEVICE_NILRT_LEGACY_TOOLCHAIN)
+  find_package(nlohmann_json REQUIRED)
+  find_library(gtest REQUIRED)
+else()
+  add_subdirectory(third_party/json ${CMAKE_CURRENT_BINARY_DIR}/json EXCLUDE_FROM_ALL)
+  add_subdirectory(third_party/gtest ${CMAKE_CURRENT_BINARY_DIR}/gtest EXCLUDE_FROM_ALL)
+endif()
 
 # Add googletest tests
 enable_testing()
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
```

Comment:
Per the Teams conversation, move these so that they're called prior to adding gtest. I think it should be OK for them to be outside of the if block.

Reply:
I think I like the idea that testing won't be enabled at all in the NILRT-OE case. I'll see if I can get it working within the branch structure.